### PR TITLE
EQL: Make queries using Point-In-Time rely on index filtering

### LINKED ISSUE #63132: EQL: PIT usage should be restricted to only the relevant filters
Currently Point-In-Time hits all indices so all queries done using PIT should be restricted to certain indices through the query not through the search request index (see https://github.com/elastic/elasticsearch/pull/63032#discussion_r498283834).

### PR DESCRIPTION
**Pull request by costin**
Point-In-Time queries cannot be ran on individual indices but on all.
Thus all PIT queries move their index from the request level to a filter
so this condition is fulfilled while keeping the query scoped
accordingly.

Fix #63132

### CODE REVIEW

File: x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/execution/search/PITAwareQueryClient.java [hunk 1]
```diff
@@ -91,6 +97,23 @@ private void searchWithPIT(MultiSearchRequest search, ActionListener<MultiSearch
...
 
+    private void makeRequestPITCompatible(SearchRequest request) {
+        SearchSourceBuilder source = request.source();
+        // don't increase the keep alive
+        source.pointInTimeBuilder(new PointInTimeBuilder(pitId));
+        // move the indices from the search request to a index filter - see #63132
+        String[] indices = request.indices();
+        if (CollectionUtils.isEmpty(indices) == false) {
+            request.indices(Strings.EMPTY_ARRAY);
+            BoolQueryBuilder indexFilter = boolQuery().filter(termsQuery(GetResult._INDEX, indices));
```
Comment (martijnvg) [inline x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/execution/search/PITAwareQueryClient.java:108]: I think `IndexFieldMapper` can't yet query against data streams. So if a data stream is provided as indices then that will correct always yield no results.

I think the `SearchIndexNameMatcher` class needs to be modified in order to take into account data streams. The 
`matchesIndex()` method should invoke the overloaded version of `concreteIndexNames()` method that accepts `includeDataStreams` parameter and provide `true`.

Comment (martijnvg) [reply x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/execution/search/PITAwareQueryClient.java:108]: Should I fix that in a separate change? I think a dedicated test for this should exists as well.

Comment (jimczi) [reply x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/execution/search/PITAwareQueryClient.java:108]: +1

Comment (jimczi) [reply x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/execution/search/PITAwareQueryClient.java:108]: Thanks for looking @martijnvg

Comment (martijnvg) [reply x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/execution/search/PITAwareQueryClient.java:108]: I've opened: #63170

**Comment by costin** [comment]
@elasticmachine update branch

**Pull request merged**